### PR TITLE
feat(historical-exports): Include historical exports finishing in activity log

### PR DESCRIPTION
**Pull request by macobo**
## Problem

More context under https://github.com/PostHog/posthog/pull/12052

Depends on https://github.com/PostHog/posthog/pull/12052 and https://github.com/PostHog/posthog/pull/12055. The changeset may be weird since it includes changes from both.

## Changes

- Add is_system column to ActivityLog db table
- Allow rendering "system" logs
- Add capability to log export finishing/failing to plugin-server exports v2
- Add tests missing from https://github.com/PostHog/posthog/pull/12052

## How did you test this code?

image

Note something is off with LetterMark colors - would ask ben but he's out sick right now.

### CODE REVIEW

File: frontend/src/scenes/plugins/pluginActivityDescriptions.tsx [hunk 1]
```diff
@@ -94,6 +95,37 @@ export function pluginActivityDescriber(logItem: ActivityLogItem): HumanizedChan
...
+
+        return {
+            description: (
+                <>
+                    Fatal error exporting historical events between {startDate} and {endDate} (inclusive). Check logs
+                    for more details.
+                </>
+            ),
+        }
+    }
```
Comment (tiina303) [inline frontend/src/scenes/plugins/pluginActivityDescriptions.tsx:127]: maybe sth like:
```suggestion
    if (logItem.activity.startsWith('export_') && logItem.detail.trigger) {
        const { dateFrom, dateTo } = logItem.detail.trigger.payload
        const startDate = dayjs(dateFrom).format('YYYY-MM-DD')
        // :TRICKY: Internally export date range is non-inclusive so transform it to be inclusive
        const endDate = dayjs(dateTo).subtract(1, 'day').format('YYYY-MM-DD')
        const description = (logItem.activity == 'export_success' ? 'Finished' : 'Fatal error') +  ' exporting historical events between {startDate} and {endDate} (inclusive).'

        return {
            description: (
                <>
                    {description}
                </>
            ),
        }
    }
```

File: posthog/models/activity_logging/activity_log.py [hunk 2]
```diff
@@ -77,6 +77,9 @@ class Meta:
     team_id = models.PositiveIntegerField(null=True)
     organization_id = models.UUIDField(null=True)
     user = models.ForeignKey("posthog.User", null=True, on_delete=models.SET_NULL)
+    # If truthy, user can be unset and this indicates a 'system' user made activity asynchronously
+    is_system = models.BooleanField(null=True)
```
Comment (tiina303) [inline posthog/models/activity_logging/activity_log.py:81]: Do we need this? Don't we already have a user/person we're using as a system user for stuff like this?

Comment (macobo) [reply posthog/models/activity_logging/activity_log.py:81]: >  Don't we already have a user/person we're using as a system user for stuff like this?

Nope

File: frontend/src/scenes/plugins/pluginActivityDescriptions.tsx (hunk 1 above)
Comment (macobo) [reply frontend/src/scenes/plugins/pluginActivityDescriptions.tsx:127]: I don't see a reason why that's better - deduplicating markdown is not a huge win.

Comment (tiina303) [reply frontend/src/scenes/plugins/pluginActivityDescriptions.tsx:127]: mainly because you were doing something //TRICKY twice

**Review by tiina303 — APPROVED** [review]
I'm still not convinced about the `is_system` field need, but let's move fast

**Pull request merged**